### ISSUE TITLE
Multiple ExcerptAppender with single ExcerptTailer issue

### ISSUE THREAD

**Issue by TangPere** [OpenHFT/Chronicle-Queue#627]
I'm using version 5.17.26

In my case.
RollCycle: Minutely
Two appender: appA, appB
One Tailer: tailer

while true to call tailer.readBytes(bytes -> System.out.println(bytes.readUtf8()));

Step 1
Time: 08:00
appA.writeBytes(bytes -> bytes.writeUtf8("hi a")) at 08:00 after write, make the appA thread sleep
appB.writeBytes(bytes -> bytes.writeUtf8("hi b")) at 08:00 after write, make the appB thread sleep

Console output:
hi a
hi b

file:
20191030-0800.cq4

-------------------------------------------------------------------------------------------------------

Step 2
Time: 0801
Wakeup appB thread and
appB.writeBytes(bytes -> bytes.writeUtf8("hi b")) at 08:01 after write, make the appB thread sleep

Console output:
hi a
hi b
**hi b**

file:
20191030-0800.cq4
**20191030-0801.cq4**

-------------------------------------------------------------------------------------------------------

Step 3
Time: 0802
Wakeup appA thread and
appA.writeBytes(bytes -> bytes.writeUtf8("hi a")) at 08:02 after write, make the appA thread sleep

Console output:
hi a
hi b
hi b
(no hi a print)

file:
20191030-0800.cq4
20191030-0801.cq4
**20191030-0802.cq4**
(But have a new file)

-------------------------------------------------------------------------------------------------------

Step 3
Time: 0803
Wakeup appB thread and
appB.writeBytes(bytes -> bytes.writeUtf8("hi B")) at 08:03 after write, make the appA thread sleep

Console output:
hi a
hi b
hi b
**hi a**(this should appA write)
(But no hi b print)

file:
20191030-0800.cq4
20191030-0801.cq4
20191030-0802.cq4
**20191030-0803.cq4**
(But have a new file)

-----------------------------------------------------------------------------------------------------

Final Step
Time: 0804
Wakeup appA and appB thread and
appA.writeBytes(bytes -> bytes.writeUtf8("hi A")) at 08:04
appB.writeBytes(bytes -> bytes.writeUtf8("hi B")) at 08:04

Console output:
hi a
hi b
hi b
hi a
**hi b**
**hi a**
**hi b**

file:
20191030-0800.cq4
20191030-0801.cq4
20191030-0802.cq4
20191030-0803.cq4
**20191030-0804.cq4**

Have you ever seen this situation?
Thanks.

**Comment by TangPere** [comment]
> @TangPere can you share the code for this?

https://gist.github.com/TangPere/0361025f21f73d3ac79e7c681b7f3beb

and the output in my pc:
```
1. a:2019-10-30T23:53:43.088413400::::2019-10-30T23:53:43.142413100
1. b:2019-10-30T23:53:46.361412500::::2019-10-30T23:53:46.362410400
2. b:2019-10-30T23:54:46.368357900::::2019-10-30T23:54:46.370362600
2. a:2019-10-30T23:55:43.093150300::::2019-10-30T23:56:46.371234900
3. b:2019-10-30T23:56:46.385211900::::2019-10-30T23:57:43.096213100
3. a:2019-10-30T23:57:43.097216700::::2019-10-30T23:57:46.387210200
4. b:2019-10-30T23:57:46.387210200::::2019-10-30T23:57:46.387210200
```

**Comment by dpisklov** [comment]
@TangPere and what exactly is wrong in this output? You write 7 messages in total, you get 7 printed, in correct order.

**Comment by TangPere** [comment]
> @TangPere and what exactly is wrong in this output? You write 7 messages in total, you get 7 printed, in correct order.

The msg write time and read time have diff, No 4 5 6 msg. The tailer should read the msg immediately, but in this demo, it have delay when using two appender write msg in diff minute.

**Comment by TangPere** [comment]
Any update

**Comment by dpisklov** [comment]
@TangPere I don't know what delay you are talking about. As your code is not really a test case I rewritten it to use secondly roll cycle as follows:
```
    private SingleChronicleQueue queue = SingleChronicleQueueBuilder.binary("queue1").rollCycle(RollCycles.TEST_SECONDLY).build();

    public void testWrite1ToQueue() {
        try {

            final ExcerptAppender appender = queue.acquireAppender();
            appender.writeBytes(bytes -> bytes.writeUtf8("1. a").writeUtf8(LocalDateTime.now().toString()));
            Thread.sleep(2 * 1000);
            appender.writeBytes(bytes -> bytes.writeUtf8("2. a").writeUtf8(LocalDateTime.now().toString()));
            Thread.sleep(2 * 1000);
            appender.writeBytes(bytes -> bytes.writeUtf8("3. a").writeUtf8(LocalDateTime.now().toString()));
        } catch (InterruptedException e) {
            e.printStackTrace();
        }
    }

    public void testWrite2ToQueue() {
        try {
            final ExcerptAppender appender = queue.acquireAppender();
            appender.writeBytes(bytes -> bytes.writeUtf8("1. b").writeUtf8(LocalDateTime.now().toString()));
            Thread.sleep(1000);
            appender.writeBytes(bytes -> bytes.writeUtf8("2. b").writeUtf8(LocalDateTime.now().toString()));
            Thread.sleep(2 * 1000);
            appender.writeBytes(bytes -> bytes.writeUtf8("3. b").writeUtf8(LocalDateTime.now().toString()));
            Thread.sleep(1000);
            appender.writeBytes(bytes -> bytes.writeUtf8("4. b").writeUtf8(LocalDateTime.now().toString()));
        } catch (InterruptedException e) {
            e.printStackTrace();
        }
    }

    public void testReadFromQueue() {
        final ExcerptTailer tailer = queue.createTailer().toEnd();
        while (!Thread.interrupted()) {
            tailer.readBytes(bytes -> {
                System.out.print(bytes.readUtf8() + ":" + bytes.readUtf8());
                System.out.println("::::" + LocalDateTime.now().toString());
            });
        }
    }

    @Test
    public void test() throws InterruptedException {
        final Thread t3 = new Thread(this::testReadFromQueue);
        t3.start();
        Thread.sleep(100);
        final Thread t1 = new Thread(this::testWrite1ToQueue);
        t1.start();
        final Thread t2 = new Thread(this::testWrite2ToQueue);
        t2.start();

        t1.join();
        t2.join();
        t3.interrupt();
        t3.join();
    }
```

This outputs
```
1. b:2019-11-05T14:06:27.778::::2019-11-05T14:06:27.780
1. a:2019-11-05T14:06:27.791::::2019-11-05T14:06:27.791
2. b:2019-11-05T14:06:28.783::::2019-11-05T14:06:28.783
2. a:2019-11-05T14:06:29.793::::2019-11-05T14:06:29.794
3. b:2019-11-05T14:06:30.785::::2019-11-05T14:06:30.786
4. b:2019-11-05T14:06:31.788::::2019-11-05T14:06:31.789
3. a:2019-11-05T14:06:31.794::::2019-11-05T14:06:31.795
```

The output is just as expected. So if you see issue please provide an actual test case with failed assertion.

**Comment by TangPere** [comment]
@dpisklov 
When new the SingleChronicleQueue on each thread will cause that issue. 
Mock it as distributed application(Same machine).
https://gist.github.com/TangPere/ea070bb74ab8fd7b53f8bf254958e530

**Comment by TangPere** [comment]
```
1. b:2019-11-05T23:37:26.360810300::::2019-11-05T23:37:26.367808200
1. a:2019-11-05T23:37:26.393808800::::2019-11-05T23:37:26.393808800
Shrunk file queue1\20191105-153726T.cq4
2. b:2019-11-05T23:37:27.378814700::::2019-11-05T23:37:27.380810200
Shrunk file queue1\20191105-153727T.cq4
2. a:2019-11-05T23:37:28.398809300::::2019-11-05T23:37:29.390810
Shrunk file queue1\20191105-153729T.cq4
Shrunk file queue1\20191105-153728T.cq4
3. b:2019-11-05T23:37:29.390810::::2019-11-05T23:37:30.406807700
4. b:2019-11-05T23:37:30.399811::::2019-11-05T23:37:30.410810500
3. a:2019-11-05T23:37:30.405809300::::2019-11-05T23:37:30.411809600
```

**Comment by dpisklov** [comment]
@TangPere  Alright, pretty funny issue on the edge case you've found - thanks for reporting, I have fixed it now.

**Comment by hft-team-city** [comment]
Released in [Chronicle-Queue-5.17.28](https://github.com/OpenHFT/Chronicle-Queue/releases/chronicle-queue-5.17.28), [BOM-2.17.428](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.17.428)
